### PR TITLE
feat(picking): implement picking lists component loading indicator

### PR DESCRIPTION
**Pull request by aleespryker**
closes: [HRZ-89545](https://spryker.atlassian.net/browse/HRZ-89545)

[HRZ-89545]: https://spryker.atlassian.net/browse/HRZ-89545?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/picking/picking-lists/picking-lists.component.ts [hunk 1]
```diff
@@ -32,6 +39,13 @@ export class PickingListsComponent extends I18nMixin(LitElement) {
 
   protected searchValue$ = new Subject<string>();
 
+  protected injector = resolve(INJECTOR);
```
Comment (supproduction) [inline libs/domain/picking/picking-lists/picking-lists.component.ts]: do we need this injector?

File: libs/domain/picking/picking-lists/picking-lists.component.ts [hunk 2]
```diff
@@ -124,15 +143,30 @@ export class PickingListsComponent extends I18nMixin(LitElement) {
...
-        ${this.i18n('picking.filter.<count>-open-pick-lists', {
-          count: this.$pickingLists()?.length ?? 0,
-        })}
-      </span>
+      ${when(
+        this.$syncing(),
+        () => html`<span class="sync"
+          ><span>${i18n('picking.loading-data')}</span>
+          <oryx-spinner></oryx-spinner
+        ></span>`,
```
Comment (supproduction) [inline libs/domain/picking/picking-lists/picking-lists.component.ts]:
```suggestion
        () => html`<span class="sync">
        <span>${i18n('picking.loading-data')}</span>
          <oryx-spinner></oryx-spinner>
          </span>`,
```

**Review by supproduction — COMMENTED** [review]
Please, change the title to `feat(picking)...`

File: libs/domain/picking/offline/services/picking-sync-action-handler.service.spec.ts [hunk 3]
```diff
@@ -195,5 +196,10 @@ describe('PickingSyncActionHandlerService', () => {
         allKeys: true,
       });
     });
+
+    it('should update syncing status', () => {
+      expect(callback).toHaveBeenCalledWith(true);
+      expect(callback).toHaveBeenCalledWith(false);
```
Comment (nsilva-spryker) [inline libs/domain/picking/offline/services/picking-sync-action-handler.service.spec.ts]: This seems ambiguous, is true and false at the same time

File: libs/domain/picking/offline/services/picking-sync-action-handler.service.ts [hunk 4]
```diff
@@ -97,6 +104,8 @@ export class PickingSyncActionHandlerService
         await store.bulkPut(pickingLists, {
           allKeys: true,
         });
+
+        this.syncing$.next(false);
```
Comment (nsilva-spryker) [inline libs/domain/picking/offline/services/picking-sync-action-handler.service.ts]: should we put it on tap, in case of error

**Review by nsilva-spryker — CHANGES_REQUESTED** [review]
Looks good. Small comments

**Review by nsilva-spryker — APPROVED** [review]

**Review by supproduction — APPROVED** [review]

**Pull request merged**